docs/site/Deploying-to-IBM-Cloud.md

```diff
@@ -0,0 +1,215 @@
+
+### Connecting the database to app
+
+Before we can use the provisioned database, we have to connect the database to
+an app on Cloud Foundry.
+
+So, create a
+[Node.js SDK app](https://console.bluemix.net/catalog/starters/sdk-for-nodejs)
+and connect it to the provisioned Cloudant service.
+
+Make a note of the name of the app. Let's say you named it `mylb4app`. We will
+be using this name as a placeholder for the unique name you will choose yourself
+eventually. Replace `mylb4app` with your app's name in all instances in the doc.
+
+## Local setup
+
+### Database
+
+1.  [Download Cloudant image](https://hub.docker.com/r/ibmcom/cloudant-developer/).
+2.  Ensure port 8080 is free and start Cloudant:
+
+```
+docker run \
+      --volume cloudant:/srv \
+      --name cloudant-developer \
+      --publish 8080:80 \
+      --hostname cloudant.dev \
+      ibmcom/cloudant-developer
+```
+
+3.  Log on to the local Cloudant Dashboard using `admin`:`pass` as the
+    crendentials.
+4.  Create a database named `todo`.
+
+### Updates to the app
+
+#### Updating datasource
+
+Update `db.datasource.json` to use the Cloudant connector.
+
+```js
+{
+  "name": "db",
+  "connector": "cloudant",
+  "url": "http://admin:pass@localhost:8080",
+  "database": "todo",
+  "modelIndex": ""
+}
+```
+
+Install the `loopback-connector-cloudant` package.
+
+```sh
+$ npm i loopback-connector-cloudant
+```
+
+#### Updating npm script
+
+Remove the `prestart` script from `package.json`, since we don't want to do any
+building on the cloud.
+
+#### Updating application
+
+We will use the `cfenv` module to simplify some of the Cloud Foundry related
+operations. Install `cfenv` in the project directory.
+
+```sh
+$ npm i cfenv
+```
+
+Update the `src/index.ts` file to the following to enable service binding.
+
+```ts
+import {TodoListApplication} from './application';
+import {ApplicationConfig} from '@loopback/core';
+const datasourceDb = require('./datasources/db.datasource.json');
+const cfenv = require('cfenv');
+const appEnv = cfenv.getAppEnv();
+
+export async function main(options?: ApplicationConfig) {
+  // Set the port assined for the app
+  if (!options) {
+    options = {rest: {port: appEnv.port}};
+  } else {
+    if (!options.rest) options.rest = {};
+    options.rest.port = appEnv.port || options.rest.port;
+  }
+
+  const app = new TodoListApplication(options);
+  // If running on IBM Cloud, we get the Cloudant service details from VCAP_SERVICES
+  if (!appEnv.isLocal) {
+    // 'myCloudant' is the name of the provisioned Cloudant service
+    const updatedDatasourceDb = Object.assign({}, datasourceDb, {
+      url: appEnv.getServiceURL('myCloudant'),
+    });
+    app.bind('datasources.config.db').to(updatedDatasourceDb);
+  }
+  await app.boot();
+  await app.start();
+
+  const url = app.restServer.url;
+  console.log(`Server is running at ${url}`);
+  return app;
+}
+```
+
+## Deploying the app
+
+### Local
```

Comment:
I'm not sure how I feel about the `local` step -- it's not necessary for a deploy of the application.

Reply:
Removed.